specs/schemes/exact/scheme_exact_algo.md

```diff
@@ -0,0 +1,226 @@
+
+#### Types
+
+Before encoding, all `txn` parameters **MUST** be converted into primitive types:
+
+- `gh`: The genesis hash of the network, which is a base64 encoded string should be decoded. 
+- `snd`: The sender address should be decoded into its public key bytes.
+- `rcv`: The receiver address should be decoded into its public key bytes.
+- `xaid`: The asset ID should be converted into a uint64 integer.
+- `aamnt`: The amount to transfer should be converted into a uint64 integer.
+- `fee`: The fee for the transaction should be converted into a uint64 integer.
+- `fv`: The first valid round should be converted into a uint64 integer.
+- `lv`: The last valid round should be converted into a uint64 integer.
+
+#### Scheme
+
+Algorand uses [**msgpack**](https://www.npmjs.com/package/algorand-msgpack) to encode the transaction before signing.
+
+
+#### Steps to sign
+
+- Decode the `txn` parameters as described above.
+- Sort fields alphabetically. 
+- Encode the sorted fields using msgpack.
+- Add a prefix `TX` in its byte representation to the encoded message.
+- Sign the resulting byte array using the sender's private key with the `Ed25519` signature scheme.
+
+
+## Full `X-PAYMENT` header:
+
+```json
+{
+  "x402Version": 1,
+  "scheme": "exact",
+  "network": "algorand-mainnet",
+  "payload": {
+    "paymentTxID": "R4PFEBJGLXV4PDOIBLY7UKBLVLXFCPYLXGBILC22BAVILTKWV3TA", // ID of the transaction in the group that will pay the resource server
+    "paymentGroup": [
+      {
+        "signature": "VxpW6emWwct7ymzyXa/F2T7rwepM71XwsslPKEdpVqmbcRUB28QMdYJuuLkeP4dnVoI7emmxT6Q/YktqHc8xBg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "fee": "2000",
+          "fv": "6631154",
+          "lv": "6632154",
+          "snd": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "rcv": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "amnt": "0",
+          "type": "pay"
+        }
+      },
+      {
+        "signature": "wVGxbZwPd2c8zKX4RSE4muFRBOHe/nRq5VT6iMeMksi2agfOCTs+VzOGUPJ+bTT5T21EuYjoBKKDX2+/NZG8Cg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "asnd": "PPNHVQJGE6ZMEWPR35UHLUYMCCZV6VNTHLJMZDVCONXKUPV47K4WGQJWGI",
+          "arcv": "ACTSMNPJPS5JMZJJ5G73JOXUUMWXXDGS7TMOER3M4W7BC54ERSZ4KPSMHM",
+          "aamt": "1000000",
+          "type": "axfer",
+          "fee": "0",
+          "fv": "6631154",
+          "lv": "6632154",
+          "xaid": "31566704"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Verification
+
+Steps to verify a payment for the `exact` scheme on Algorand:
+
+1. Verify the signature is valid for each transaction in the `paymentGroup`.
+  a. Encode the `txn` parameters as described in the encoding section.
+  b. Recover the public key of the sender from the `snd` or `asnd` field by decoding from the address.
```

Comment:
This is not always true if we use a multi-sig or a logic-sig.

Reply:
`snd` should always the one signing, `asnd` is just _who_ the assets are coming from. But moreover we haven't considered rekeyed accounts here. The signature might be valid for the sender, but if the account has defined a different auth-addr, then it'd fail when it gets submitted to the network. I would probably suggest we reduce this section down to just step 3? Since the simulation endpoint can cover step 1 and 2 anyway?
```suggestion
  b. Recover the public key of the sender from the `snd` field by decoding from the address.
```